plutus-tx/src/PlutusTx/Lift.hs

```diff
@@ -70,9 +70,10 @@ safeLift x = do
     tcConfig <- PLC.getDefTypeCheckConfig $ Original ()
     -- NOTE:  Disabling simplifier, as it takes a lot of time during runtime
     let ccConfig = set (ccOpts . coMaxSimplifierIterations) 0 (toDefaultCompilationCtx tcConfig)
+        usOpts = set UPLC.soMaxSimplifierIterations 0 UPLC.defaultSimplifyOpts
     compiled <- flip runReaderT ccConfig $ compileTerm lifted
     let erased = UPLC.erase compiled
-    db <- UPLC.deBruijnTerm $ UPLC.simplifyTerm erased
+    db <- UPLC.deBruijnTerm =<< UPLC.simplifyTerm usOpts erased
```

Comment:
Great question. It was there already :laughing: 

Lifting terms *absolutely* produces stuff that needs simplifying: it just runs the compiler! So really we *should* simplify here otherwise we're leaving lots of garbage for people. But then we're simplifying at runtime.

I'm increasingly uncomfortable with `Lift`, partly for these reasons... :cry: